spark/v3.4/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/CreateOrReplaceTagExec.scala

```diff
@@ -50,15 +51,19 @@ case class CreateOrReplaceTagExec(
           "Cannot complete create or replace tag operation on %s, main has no snapshot", ident)
 
         val manageSnapshot = iceberg.table.manageSnapshots()
-        if (!replace) {
+        if (create && replace && null == iceberg.table().refs().get(tag)) {
```

Comment:
This is a little hard to reason through, what about simplifying to something more like:

```scala
val exists = null == iceberg.table().refs().get(tag)

if (replace && exists) {
  manageSnapshot.replaceTag(tag, snapshotId)
} else if (create && !exists) {
  if (exists && ifNotExists) {
    return Nil
  }
  manageSnapshot.createTag(tag, snapshotId)
}
```

Reply:
that will change the expected behavior unfortunaly. Tests are expecting `ALTER TABLE x REPLACE TAG non-existing` to throw an exception, therefore we need to distinguish whether `CREATE OR REPLACE` is used here vs `REPLACE`

Reply:
yes, this would restore the behavior I mentioned above, but it would change a different behavior where `ALTER TABLE x CREATE TAG t` is expected to fail when `t` already exists. The conditions are quite tricky unfortunately and from what I've seen is that we need to explicitly handle `create && replace && !refExists`